core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCnaDiscreteLongData.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2018 - 2022 The Hyve B.V.
+ * This code is licensed under the GNU Affero General Public License (AGPL),
+ * version 3, or (at your option) any later version.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.mskcc.cbio.portal.scripts;
+
+import com.google.common.base.*;
+import com.google.common.collect.*;
+import org.mskcc.cbio.portal.dao.*;
+import org.mskcc.cbio.portal.model.*;
+import org.mskcc.cbio.portal.util.*;
+
+import java.io.*;
+import java.util.Optional;
+import java.util.*;
+import java.util.stream.*;
+
+import static com.google.common.collect.Lists.*;
+import static java.lang.String.*;
+
+public class ImportCnaDiscreteLongData {
+
+    private final File cnaFile;
+    private final int geneticProfileId;
+    private final GeneticAlterationImporter geneticAlterationGeneImporter;
+    private String genePanel;
+    private final DaoGeneOptimized daoGene;
+    private CnaUtil cnaUtil;
+    private Map<CnaEvent.Event, CnaEvent.Event> existingCnaEvents = new HashMap<>();
```

Comment:
I thought the same thing - it's because he wants to retrieve the existing cna event from the map when he finds an equivalent one, and replace the reference to the equivalent one with the original.

Reply:
I reused old functionality: after a closer look, I think we can replace the map with a set. 

(The only real difference is that we now need to find the stored event in the Set, but this works fine: both a hashmap and a hashset should compare by event.hashcode and event.equals, which compare by gene and alteration.)

See [884fca6](https://github.com/cBioPortal/cbioportal/pull/9847/commits/884fca60ab44ac04dbeb27800587c0e354e54f0f).

All tests are still working.